core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -117,658 +148,904 @@ public boolean isPrettyPrint() {
             startArray();
-            for (Object o : value) {
-                value(o);
+            for (Object value : values) {
+                unknownValue(value);
             }
             endArray();
         }
         return this;
     }
 
-    public XContentBuilder latlon(String name, double lat, double lon) throws IOException {
-        return startObject(name).field("lat", lat).field("lon", lon).endObject();
+    ////////////////////////////////////////////////////////////////////////////
+    // Misc.
+    //////////////////////////////////
+
+    public XContentBuilder timeValueField(String rawFieldName, String readableFieldName, TimeValue timeValue) throws IOException {
+        if (humanReadable) {
+            field(readableFieldName, timeValue.toString());
+        }
+        field(rawFieldName, timeValue.millis());
+        return this;
     }
 
-    public XContentBuilder latlon(double lat, double lon) throws IOException {
-        return startObject().field("lat", lat).field("lon", lon).endObject();
+    public XContentBuilder timeValueField(String rawFieldName, String readableFieldName, long rawTime) throws IOException {
+        if (humanReadable) {
+            field(readableFieldName, new TimeValue(rawTime).toString());
+        }
+        field(rawFieldName, rawTime);
+        return this;
+    }
+
+    public XContentBuilder timeValueField(String rawFieldName, String readableFieldName, long rawTime, TimeUnit timeUnit) throws
+            IOException {
+        if (humanReadable) {
+            field(readableFieldName, new TimeValue(rawTime, timeUnit).toString());
+        }
+        field(rawFieldName, rawTime);
+        return this;
+    }
+
+
+    public XContentBuilder percentageField(String rawFieldName, String readableFieldName, double percentage) throws IOException {
+        if (humanReadable) {
+            field(readableFieldName, String.format(Locale.ROOT, "%1.1f%%", percentage));
+        }
+        field(rawFieldName, percentage);
+        return this;
+    }
+
+    public XContentBuilder byteSizeField(String rawFieldName, String readableFieldName, ByteSizeValue byteSizeValue) throws IOException {
+        if (humanReadable) {
+            field(readableFieldName, byteSizeValue.toString());
+        }
+        field(rawFieldName, byteSizeValue.bytes());
+        return this;
+    }
+
+    public XContentBuilder byteSizeField(String rawFieldName, String readableFieldName, long rawSize) throws IOException {
+        if (humanReadable) {
+            field(readableFieldName, new ByteSizeValue(rawSize).toString());
+        }
+        field(rawFieldName, rawSize);
+        return this;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    //
+    // Methods that actually write to the underlying XContentGenerator
+    //
+    ////////////////////////////////////////////////////////////////////////////
+
+    private void writeStartObject() throws IOException {
```

Comment:
> While you are cleaning up maybe we should put these into their own class that XContentBuilder extends? That way all of the wrapper methods are all in one spot.

I'd prefer not: these methods already delegate to XContentGenerator which delegates to JsonGenerator etc, so I prefer to keep this simple.

Since all the methods have been grouped by types and correctly delegate to each other, I think there's no more need to group the writing method separate. We could just merge them with their calling method and have:

```
public XContentBuilder value(boolean value) throws IOException {
        generator.writeBoolean(value);
        return this;
}
```

instead of 

```
public XContentBuilder value(boolean value) throws IOException {
        writeBoolean(value);
        return this;
}

private void writeBoolean(boolean value) throws IOException {
        generator.writeBoolean(value);
        return this;
}
```

We'll just have to take care when adding new methods that we don't fall back in the previous "let's duplicate everything" style.